components/board/customlinux/board.go

```diff
@@ -0,0 +1,114 @@
+//go:build linux
+
+// Package customlinux implements a board running linux
+package customlinux
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"strconv"
+
+	"github.com/edaniels/golog"
+	"periph.io/x/host/v3"
+
+	"go.viam.com/rdk/components/board"
+	"go.viam.com/rdk/components/board/genericlinux"
+	"go.viam.com/rdk/resource"
+)
+
+func init() {
+	if _, err := host.Init(); err != nil {
+		golog.Global().Debugw("error initializing host", "error", err)
+	}
+
+	RegisterCustomBoard("customlinux")
+}
+
+type customSysFsBoard struct {
+	*genericlinux.SysfsBoard
+}
+
+func (b *customSysFsBoard) Reconfigure(
+	ctx context.Context,
+	_ resource.Dependencies,
+	conf resource.Config,
+) error {
+	newConf, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return err
+	}
+
+	return b.ReconfigureParsedConfig(ctx, &newConf.BoardConfig)
+}
+
+// RegisterCustomBoard registers a sysfs based board using the pin mappings.
+func RegisterCustomBoard(modelName string) {
+	resource.RegisterComponent(
+		board.API,
+		resource.DefaultModelFamily.WithModel(modelName),
+		resource.Registration[board.Board, *Config]{
+			Constructor: func(
+				ctx context.Context,
+				_ resource.Dependencies,
+				conf resource.Config,
+				logger golog.Logger,
+			) (board.Board, error) {
+				newConf, err := resource.NativeConfig[*Config](conf)
+				if err != nil {
+					return nil, err
+				}
+
+				pinData, err := os.ReadFile(newConf.PinConfigFilePath)
+				if err != nil {
+					return nil, err
+				}
+				var parsedPinData GenericLinuxPins
+				if err := json.Unmarshal(pinData, &parsedPinData); err != nil {
+					return nil, err
+				}
+				pinDefs := make([]genericlinux.PinDefinition, len(parsedPinData.Pins))
+
+				var pinName int
+				var pwmID int
+				for i, pin := range parsedPinData.Pins {
+					pinName, err = strconv.Atoi(pin.Name)
+					if err != nil {
+						return nil, err
+					}
+					if pin.PWMID == "" {
+						pwmID = -1
+					} else {
+						pwmID, err = strconv.Atoi(pin.PWMID)
```

Comment:
This is coming from a brand new data format you've defined. There should be no need to convert it to a different type; just define the one you want.

If you did this because you want omitted values to become -1 and not 0, all we need to do is define the `UnmarshalJSON` function on your struct to give a default value, I think. See https://stackoverflow.com/a/41102996 for details